Added SageMaker extension and converted templates to YAML

mgawino opened the pull request:
Added demonstration of SageMaker model building and deployment. All Cloud Formation templates were converted from JSON to YAML format. Taskcat tests passed. The key changes are in templates/sagemaker.yaml.

Please review, thank you!

sshvans commented on templates/iam-roles.yaml:
```diff
@@ -0,0 +1,958 @@
+              - Effect: Allow
+                Action:
+                  - s3:*
+                Resource: arn:aws:s3:::*
```
Does this role need access to s3:::* ? Can we restrict it to specific bucket?

mgawino commented:
@sshvans I've fixed roles for SageMaker instance and also added a few improvements. SnsLearnMoreTopicArn is used for gathering messages sent on the last page of wizard (learn more about 47Lining's assessment offering).
